File: dotnet/Workspace.Azure.Storage/Account.cs
using YamlDotNet.Serialization;
using Microsoft.WindowsAzure.Storage;

namespace Microsoft.Workspace.Azure.Storage
{
    public class Account : Resource
    {
        [YamlMember(Alias = "accountname")]
        public string Name { get; set; }

        public CloudStorageAccount Client
        {
            get
            {
                // TODO: lookup subscriptions
                // search for storage account
                // TODO: lookup keyvault
                // TODO: support service prinicipal
                // TODO: support env variable
                // CloudStorageAccount.Parse
                return null;
            }
        }
    }
}

File: typescript/src/azure/storage/account.ts
import Resource from "../../Resource";
import ResourceType from "../../ResourceType";

@ResourceType('azure.storage.account')
export default class AzureStorageAccount extends Resource {
	constructor()
	constructor(public accountname?: string) {
		super();
	}

	getClient() {
		console.log("get_client invoked")
	}

}